### ISSUE TITLE
ENH: Options to change color of legend's text in gdf.explore() 

### ISSUE THREAD

**Issue by gcaria** [geopandas/geopandas#2897]
#### Is your feature request related to a problem?

I wish I could use `explore` with a colormap also on tiles where dense forests have dark colors, and I can't read the colorbar's title and ticks since they are written with a black font.

#### Describe the solution you'd like

To be able to change color of legend's text in `gdf.explore()`, by adding a new key to `legend_kwds` 

#### API breaking implications

None, just the addition of a key (e.g. `textColor`) in `legend_kwds`.

**Comment by martinfleis** [comment]
I don't think this is supported by `branca` package we use to render colormap legends. It would need to be exposed there first and then here. Can you open an issue there?

**Comment by shriv** [comment]
@martinfleis `branca` now has this option merged (as at version 0.8.0). Would `geopandas` be able to support this feature now? Thanks!

**Comment by martinfleis** [comment]
We'll need to treat the `text_color` in the same way we treat `max_labels` here and add pop it from `legend_kwds` and pass to `colormap_kwds`.

https://github.com/geopandas/geopandas/blob/7962e97cb87b34c3c4da7b8dcd87f719c425273e/geopandas/explore.py#L684-L686

So nothing blocks it. If you'd like to make a PR implementing it, go ahead.
